From 19439e78159a1f3d7d9ba64bcfc931b3c1337f9a Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Sat, 06 Feb 2021 14:02:59 +0200
Subject: [PATCH] [SW-20154] make efuse settings platform specific

eFUSE bit definitions differ from Goya2 vs Gaudi2.
Hence make them platform specific.

Change-Id: I29c6462ccaff58c45a814132bee0c109edc8a690
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/common/efuse.h b/common/efuse.h
index 8ce9f7f..2915380 100644
--- a/common/efuse.h
+++ b/common/efuse.h
@@ -9,6 +9,7 @@
 #define _EFUSE_H_
 
 #include <stdint.h>
+#include <dev_efuse.h>
 
 #define EFUSE_READ_DONE				0x2
 #define EFUSE_NOT_READ				0x1
@@ -19,7 +20,6 @@
 #define EFUSE_SYSCFG_OFFSET			0x3A
 #define EFUSE_SECURITY_FW_VERSION_OFFSET	0x3B
 
-#define EFUSE_SYSCFG_ENABLE_4B_TERMINATION	BIT(25)
 #define EFUSE_SYSCFG_GPIO02_RECOVERY_CONTROL	BIT(24)
 #define EFUSE_SYSCFG_WA_SNPS_CASE_01038891	BIT(23)
 #define EFUSE_SYSCFG_DISABLE_ROM_PCIE_RECOVERY	BIT(22)
@@ -29,26 +29,9 @@
 #define EFUSE_SYSCFG_PCIE_GEN_LIMIT		BIT(18)
 #define EFUSE_SYSCFG_SPI_CONTROLLER		BIT(17)
 #define EFUSE_SYSCFG_DISABLE_JTAG_IF_MASK	BIT(12)
-#define EFUSE_SYSCFG_FW_VERSION_LOCATION	13
-#define EFUSE_SYSCFG_FW_VERSION_LOCATION_MASK	0x3
 #define EFUSE_SYSCFG_ENABLE_VALIDATION_MASK	BIT(10)
-#define EFUSE_SYSCFG_KEY_LOCATION		7
-#define EFUSE_SYSCFG_KEY_LOCATION_MASK		0x3
-#define EFUSE_SYSCFG_REVOKATION_LOCATION	5
-#define EFUSE_SYSCFG_REVOKATION_LOCATION_MASK	0x3
 #define EFUSE_SYSCFG_REVOKE_KEY_LOCATION	0
 #define EFUSE_SYSCFG_REVOKE_KEY_MASK		0x1F
-
-#define EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI	BIT(14)
-#define EFUSE_SYSCFG_KEY_LOCATION_SPI		BIT(8)
-#define EFUSE_SYSCFG_REVOKATION_LOCATION_SPI	BIT(6)
-
-#define EFUSE_SYSCFG_DEFAULT	(EFUSE_SYSCFG_SPI_CONTROLLER | \
-				 EFUSE_SYSCFG_ENABLE_VALIDATION_MASK | \
-				 EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI | \
-				 EFUSE_SYSCFG_KEY_LOCATION_SPI | \
-				 EFUSE_SYSCFG_REVOKATION_LOCATION_SPI | \
-				 EFUSE_SYSCFG_ENABLE_4B_TERMINATION)
 
 int fuse_init(void);
 int fuse_read(uint32_t word, uint32_t *val);
diff --git a/gaudi2_arc_gnu/dev_efuse.h b/gaudi2_arc_gnu/dev_efuse.h
new file mode 100644
index 0000000..056db38
--- /dev/null
+++ b/gaudi2_arc_gnu/dev_efuse.h
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2021 HabanaLabs Ltd.
+ */
+
+#define EFUSE_SYSCFG_RECOVERY_CODE_LOCATION		13
+#define EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK	0x1
+#define EFUSE_SYSCFG_ENABLE_TPM_MASK			BIT(11)
+#define EFUSE_SYSCFG_FW_VERSION_LOCATION		7
+#define EFUSE_SYSCFG_FW_VERSION_LOCATION_MASK		0x3
+#define EFUSE_SYSCFG_KEY_LOCATION			5
+#define EFUSE_SYSCFG_KEY_LOCATION_MASK			0x3
+#define EFUSE_SYSCFG_REVOKATION_LOCATION		7
+#define EFUSE_SYSCFG_REVOKATION_LOCATION_MASK		0x3
+
+#define EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI	BIT(8)
+#define EFUSE_SYSCFG_KEY_LOCATION_ROM		BIT(5)
+#define EFUSE_SYSCFG_REVOKATION_LOCATION_SPI	BIT(8)
+
+#define EFUSE_SYSCFG_DEFAULT	(EFUSE_SYSCFG_SPI_CONTROLLER | \
+				 EFUSE_SYSCFG_ENABLE_VALIDATION_MASK | \
+				 EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI | \
+				 EFUSE_SYSCFG_KEY_LOCATION_ROM | \
+				 EFUSE_SYSCFG_REVOKATION_LOCATION_SPI)
diff --git a/gaudi2_arc_gnu/onix.c b/gaudi2_arc_gnu/onix.c
index f82fc69..a3ce726 100644
--- a/gaudi2_arc_gnu/onix.c
+++ b/gaudi2_arc_gnu/onix.c
@@ -11,7 +11,7 @@
 
 #define EFUSE_SYSCFG_ONIX_DEFAULT \
 				(EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI | \
-				 EFUSE_SYSCFG_KEY_LOCATION_SPI | \
+				 EFUSE_SYSCFG_KEY_LOCATION_ROM | \
 				 EFUSE_SYSCFG_REVOKATION_LOCATION_SPI | \
 				 EFUSE_SYSCFG_ENABLE_VALIDATION_MASK | \
 				 EFUSE_SYSCFG_GPIO02_RECOVERY_CONTROL)
diff --git a/goya2_arc/dev_efuse.h b/goya2_arc/dev_efuse.h
new file mode 100644
index 0000000..bc1caf5
--- /dev/null
+++ b/goya2_arc/dev_efuse.h
@@ -0,0 +1,23 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2021 HabanaLabs Ltd.
+ */
+
+#define EFUSE_SYSCFG_ENABLE_4B_TERMINATION	BIT(25)
+#define EFUSE_SYSCFG_FW_VERSION_LOCATION	13
+#define EFUSE_SYSCFG_FW_VERSION_LOCATION_MASK	0x3
+#define EFUSE_SYSCFG_KEY_LOCATION		7
+#define EFUSE_SYSCFG_KEY_LOCATION_MASK		0x3
+#define EFUSE_SYSCFG_REVOKATION_LOCATION	5
+#define EFUSE_SYSCFG_REVOKATION_LOCATION_MASK	0x3
+
+#define EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI	BIT(14)
+#define EFUSE_SYSCFG_KEY_LOCATION_SPI		BIT(8)
+#define EFUSE_SYSCFG_REVOKATION_LOCATION_SPI	BIT(6)
+
+#define EFUSE_SYSCFG_DEFAULT	(EFUSE_SYSCFG_SPI_CONTROLLER | \
+				 EFUSE_SYSCFG_ENABLE_VALIDATION_MASK | \
+				 EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI | \
+				 EFUSE_SYSCFG_KEY_LOCATION_SPI | \
+				 EFUSE_SYSCFG_REVOKATION_LOCATION_SPI | \
+				 EFUSE_SYSCFG_ENABLE_4B_TERMINATION)
